### ISSUE TITLE
Download of Data produces Validation Errors (Data Checksum?)

### ISSUE THREAD

**Issue by Lucew** [alan-turing-institute/TCPD#19]
First of all, thank you for your effort in creating a change point detection benchmark. I really appreciate this as the benchmarking of algorithms is not standardized in most publications. Therefore, I wanted to use your dataset to test different algorithms.

I used a virtual environment to install all required packages as described in the Readme. Then I use
`python build_tcpd.py -v collect` within the activated virtual environment to start the download of data.

After downloading the first files for the Apple dataset (AAPL.csv and apple.json appear in the directory ./datasets/apple). The script throws and Validation error:
<details>

```
Running collect action for dataset: apple ... Traceback (most recent call last):
  File "C:\Users\Lucas\PycharmProjects\TCPD\build_tcpd.py", line 115, in <module>
    main()
  File "C:\Users\Lucas\PycharmProjects\TCPD\build_tcpd.py", line 110, in main
    func(name, script)
  File "C:\Users\Lucas\PycharmProjects\TCPD\build_tcpd.py", line 85, in collect_dataset
    return run_dataset_func(name, script, "collect")
  File "C:\Users\Lucas\PycharmProjects\TCPD\build_tcpd.py", line 81, in run_dataset_func
    func(output_dir=dir_path)
  File "./datasets\apple\get_apple.py", line 213, in collect
    write_patch(json_path, target_path=json_path)
  File "./datasets\apple\get_apple.py", line 68, in wrapper
    raise ValidationError(target)
tcpd.apple.ValidationError: ./datasets\apple\apple.json
```

</details>

If I delete the Apple Dataset from `TARGETS` in build_tcpd.py, it then proceeds with everything for bee_waggle_6. There it also notes, that the checksum does not match. The script then throws the next Validation error at bitcoin.

<details>

```
Running collect action for dataset: bee_waggle_6 ... Warning: Generated dataset ./datasets\bee_waggle_6\bee_waggle_6.json didn't match a known checksum. This is likely due to rounding differences caused by different system architectures. Minor differences in algorithm performance can occur for this dataset. 
ok
Running collect action for dataset: bitcoin ... Traceback (most recent call last):
  File "C:\Users\Lucas\PycharmProjects\TCPD\build_tcpd.py", line 114, in <module>
    main()
  File "C:\Users\Lucas\PycharmProjects\TCPD\build_tcpd.py", line 109, in main
    func(name, script)
  File "C:\Users\Lucas\PycharmProjects\TCPD\build_tcpd.py", line 84, in collect_dataset
    return run_dataset_func(name, script, "collect")
  File "C:\Users\Lucas\PycharmProjects\TCPD\build_tcpd.py", line 80, in run_dataset_func
    func(output_dir=dir_path)
  File "./datasets\bitcoin\get_bitcoin.py", line 131, in collect
    write_json(csv_path, target_path=json_path)
  File "./datasets\bitcoin\get_bitcoin.py", line 66, in wrapper
    raise ValidationError(target)
tcpd.bitcoin.ValidationError: Validating the file './datasets\bitcoin\bitcoin.json' failed.
Please raise an issue on the GitHub page for this project if the error persists.

```

</details>

Could you help me out, what is going wrong here? I would really appreciate it. Maybe this has something to do with the validation checksum computation, which might be different on my PC. The venv is using Python 3.9.13 on a Windows machine.

Best regards
Lucas

**Comment by Lucew** [comment]
I just checked the closed Issues and #3 seems to be similar to mine. So the recommendation is then to use the docker?
My guess is that some windows related character encodings, e.g. EOF, probably mess up the validation check sum.

Might that be right?

**Comment by GjjvdBurg** [comment]
Hi Lucas, thanks for opening this issue. I think your conclusion is correct: Windows-related character encodings and line endings might be the root cause of the validation errors. The dataset is [regularly checked](https://github.com/alan-turing-institute/TCPD/actions/workflows/validate.yml) using builds for docker and ubuntu, and these checks seem to pass. If you're happy to use docker on Windows then I think that's the best solution.
